src/harps-n/io.jl

```diff
@@ -0,0 +1,142 @@
+    keys = propertynames(df_files)
+    allowmissing!(df_files, keys[map(k->k∉[:Filename, :bjd, :target, :airmass],keys)] )
+
+    if length(df_filenames.Filename) >= 2
+        map(fn->add_metadata_from_fits!(df_files,joinpath(fn,"s2d.fits")),df_filenames.Filename[2:end])
+    end
+
+    RV_data = CSV.read(joinpath(data_path,"harpn_sun_release_timeseries_2015-2018.csv"), DataFrame)
+
+    # berv = RV_data[:, [:berv]]
+    berv_bary_to_helio = RV_data[:, [:berv_bary_to_helio]] .* -1.0
+    rv_diff_extinction = RV_data[:, [:rv_diff_extinction]] .* -1.0
+
+    df_files = hcat(df_files, berv_bary_to_helio, rv_diff_extinction)
+    append!(keys,[:ssb_rv, :diff_ext_rv])
+    rename!(df_files,keys)
+
+    df_files
+
+end
+
+"""Create Dict containing filename and default metadata from file."""
+function read_metadata(fn::Union{String,FITS})
+    fields_to_save = metadata_symbols_default(HARPSN2D())
+    fields_str_to_save = metadata_strings_default(HARPSN2D())
+    dict = read_metadata_from_fits(fn,fields=fields_to_save,fields_str=fields_str_to_save)
+    check_metadata_fields_expected_present(dict,fields_to_save)
+    dict[:Filename] = fn
+    if haskey(dict,:airmass) && typeof(dict[:airmass]) == String
+       dict[:airmass] = parse(Float64,dict[:airmass])
+    end
+    return dict
+end
+
+function add_metadata_from_fits!(df::DataFrame, fn::String)
+    println("# Reading metadata from ", fn)
+    metadata_keep = read_metadata(fn)
+    #reject file from df if it is a calibration (only keep spectra)
+    #if metadata_keep[:target] != "Solar" return df end
+    #calibration_target_substrings = ["Etalon","LFC","ThArS","LDLS","FlatBB"]
+    #if any(map(ss->contains( metadata_keep[:target],ss),calibration_target_substrings)) 
+    #   return df
+    #end
+
+    for k in propertynames(df)
+        if typeof(metadata_keep[k]) == Missing continue end
+        if typeof(metadata_keep[k]) == Nothing
+            metadata_keep[k] = missing
+            continue
+        end
+        if (eltype(df[!,k]) <: Union{Real,Union{<:Real,Missing}}) && !(typeof(metadata_keep[k]) <: Union{Real,Union{<:Real,Missing}})
+            metadata_keep[k] = missing
+        end
+    end
+    #println("metadata_keep = ", metadata_keep)
+    if haskey(metadata_keep,:airmass) && typeof(metadata_keep[:airmass]) == String
+       metadata_keep[:airmass] = parse(Float64,metadata_keep[:airmass])
+    end
+
+    push!(df, metadata_keep)
+    return df
+end
+
+
+""" Read HARPS-N data from FITS file, and return in a Spectra2DBasic object."""
+function read_data   end
+
+function read_data(f::FITS, metadata::Dict{Symbol,Any} )
+    @assert length(f) >= 2
+    @assert all(map(i->typeof(f[i]) <: FITSIO.ImageHDU,2:length(f)))
+    img_idx = Dict(map(i->first(read_key(f[i],"EXTNAME"))=>i,2:length(f)))
+    λ, flux, var  = FITSIO.read(f[img_idx["WAVEDATA_AIR_BARY"]]), FITSIO.read(f[img_idx["SCIDATA"]]), FITSIO.read(f[img_idx["ERRDATA"]])
```

Comment:
Did you check that ERRDATA is variance and not a standard deviation?

Reply:
Errdata is STD, so should square it